tensorboard/webapp/metrics/store/metrics_store_internal_utils_test.ts

```diff
@@ -742,4 +742,41 @@ describe('metrics store utils', () => {
       ).toEqual([10, 20, 30]);
     });
   });
+
+  describe('getSelectedSteps', () => {
+    it(`gets one selected step on single selection`, () => {
+      const selectedTime = {start: {step: 10}, end: null};
+      const stepValues = [10];
+
+      expect(getSelectedSteps(selectedTime, stepValues)).toEqual([10]);
+    });
+
+    it(`gets selected steps on range selection`, () => {
+      const selectedTime = {start: {step: 10}, end: {step: 40}};
+      const stepValues = [5, 10, 20, 40];
+
+      expect(getSelectedSteps(selectedTime, stepValues)).toEqual([10, 20, 40]);
```

Comment:
This test gives me confidence that we only select steps above the bottom of the selectedTime but NOT that we only select steps below the top of the selectedTime. Maybe add one more case?

Reply:
Change this case to steps=[5..50], time=[10,40]
and added another one for steps=[5..40], time=[10,50]